panels/models.py

```diff
@@ -0,0 +1,81 @@
+from panels import *    
+
+
+@Session.model_mixin
+class Attendee:
+    assigned_panelists = relationship('AssignedPanelist', backref='attendee')
+
+
+class Event(MagModel):
+    location    = Column(Choice(c.EVENT_LOCATION_OPTS))
+    start_time  = Column(UTCDateTime)
+    duration    = Column(Integer)   # half-hour increments
+    name        = Column(UnicodeText, nullable=False)
+    description = Column(UnicodeText)
+
+    assigned_panelists = relationship('AssignedPanelist', backref='event')
+
+    @property
+    def half_hours(self):
+        half_hours = set()
+        for i in range(self.duration):
+            half_hours.add(self.start_time + timedelta(minutes=30 * i))
+        return half_hours
+
+    @property
+    def minutes(self):
+        return (self.duration or 0) * 30
+
+    @property
+    def start_slot(self):
+        if self.start_time:
+            return int((self.start_time_local - c.EPOCH).total_seconds() / (60 * 30))
+
+
+class AssignedPanelist(MagModel):
```

Comment:
It's a little unusual to see a relational table in a postgres DB - what was the reason for doing this instead of a many-to-many relationship?